Frontend/src/Components/Dashboard/GraphContainer.js

```diff
@@ -0,0 +1,253 @@
+import { Select, useConst, VStack } from "@chakra-ui/react";
+import { useCallback, useMemo, useState } from "react";
+import CustomGraph from "../Graph/CustomGraph";
+import GraphData from "../Graph/graph-data.json";
+
+/**
+ * Generates color data for all categories/datasets and packs the data nicely
+ *
+ * The data is in this format:
+ * ```
+ * [
+ * [
+ *   {
+ *     category: string,
+ *     values: [
+ *       {
+ *         key: string,
+ *         name: string,
+ *         color: string
+ *       }
+ *     ]
+ *   }
+ * ],
+ * [
+ *   {
+ *     key: string,
+ *     name: string,
+ *     color: string
+ *   }
+ * ]
+ * ]```
+ *
+ * where `category` is the name of a category,
+ * `key` is the internally-used string used to identify the dataset,
+ * `name` is the human-readable name of this dataset, and `color` is the
+ * (CSS-compatible) hex color code assigned to this dataset.
+ *
+ * @returns a list of categories taken from the JSON file with randomly generated colors
+ */
+function generateCategories() {
+  const categories = [],
+    allDatasets = [];
+  for (const category of GraphData.categories) {
+    const values = category.values.map((obj) => {
+      const colorNum = Math.floor(Math.random() * 0x3fffff + 0x3fffff);
+      const color = "#" + colorNum.toString(16);
+
+      const output = { key: obj.key, name: obj.name, color };
+      allDatasets.push(output);
+      // console.log("Generated color for", obj, ":", output);
+      return output;
+    });
+    categories.push({ category: category.category, values });
+  }
+
+  return [categories, allDatasets];
+}
+
+/**
+ * Creates a new GraphContainer component
+ *
+ * @param {any} props the props to pass to this graph container
+ * @returns the graph-containing component
+ */
+export default function GraphContainer({ queue, latestTime, ...props }) {
+  // fetch constants: categories and flatpacked categories
+  const [categories, allDatasets] = useConst(generateCategories);
```

Comment:
Why do you need categories? It's passed to CustomGraph but isn't used in it. Was this meant to be passed into GraphSelectModal?

Reply:
I guess I don't. Removed